src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/SerializationContextTests.cs

```diff
@@ -446,6 +455,35 @@ void RunTest(ClassWithNullableProperties expected)
             }
         }
 
+#if NET6_0_OR_GREATER
+        [Fact]
+        public override void ClassWithDateOnlyAndTimeOnlyValues_Roundtrip()
+        {
+            RunTest(new ClassWithDateOnlyAndTimeOnlyValues
+            {
+                DateOnly = DateOnly.Parse("2022-05-10"),
+                NullableDateOnly = DateOnly.Parse("2022-05-10"),
+
+                TimeOnly = TimeOnly.Parse("21:51:51"),
```

Comment:
Perhaps I missed it but are there any test cases testing values like `-1:59:59`, `25:00:00`, `14:61:00`, `4:-1:0` etc?

Reply:
(at least make sure we throw the right exception)